tests/tests_release_notes_workflow/test_releasenotes_check_functions.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import unittest
+from typing import Any
+from unittest.mock import patch
+
+from git.git_repository import GitRepository
+from manifests.input_manifest import InputComponentFromSource, InputManifest
+from release_notes_workflow.component_release_notes import ComponentsReleaseNotes
+from release_notes_workflow.release_notes import ReleaseNotes
+
+
+class TestReleaseNotesCheckFunctions(unittest.TestCase):
+
+    @patch('subprocess.check_output', return_value='c3408f34961eb7e8ebc0510306f62238a457dbbc'.encode())
+    @patch("subprocess.check_output", return_value='.release-notes-2.2.0.0.md'.encode())
+    def test_plugin_check(self, *mocks: Any) -> None:
+        component = InputComponentFromSource({"name": "common-utils", "repository": "https://github.com/opensearch-project/common-utils", "ref": "c3408f34961eb7e8ebc0510306f62238a457dbbc"})
+        repo = GitRepository(
```

Comment:
I am pretty sure that this calls `__checkout__` [here](https://github.com/opensearch-project/opensearch-build/blob/main/src/git/git_repository.py#L44) and actually checks out code. Put dummy values in `url` and `ref` to ensure that this is correctly mocked.

I think you're missing another `@patch('subprocess.check_call', return_value=0)`.

Reply:
Looks like you fixed that in a subsequent commit.